h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -283,79 +282,72 @@ void checkAndInheritModelProperties() {
-          throw new H2OIllegalArgumentException("Base models are inconsistent: they use different response columns." +
-              " Found: "+responseColumn+"(StackedEnsemble) and "+aModel._parms._response_column+" (model "+k+").");
-        
-//        if (blending_mode && _parms._blending.equals(aModel._parms._train)) {
-//          throw new H2OIllegalArgumentException("Base model `"+k+"` was trained with the StackedEnsemble blending frame.");
-//        }
-
-        // TODO: we currently require xval; loosen this iff we add a separate holdout dataset for the ensemble
-        if (cv_required_on_base_model) {
-          if (aModel._parms._fold_assignment != basemodel_fold_assignment) {
-            if ((aModel._parms._fold_assignment == AUTO && basemodel_fold_assignment == Random) ||
-                (aModel._parms._fold_assignment == Random && basemodel_fold_assignment == AUTO)) {
-              // A-ok
-            } else {
-              throw new H2OIllegalArgumentException("Base models are inconsistent: they use different fold_assignments.");
-            }
-          }
-
-          // If we have a fold_column make sure nfolds from base models are consistent
-          if (aModel._parms._fold_column == null && basemodel_nfolds != aModel._parms._nfolds)
-            throw new H2OIllegalArgumentException("Base models are inconsistent: they use different values for nfolds.");
+          throw new H2OIllegalArgumentException("Base models are inconsistent: they use different response columns."
+                  +" Found: "+responseColumn+"(StackedEnsemble) and "+aModel._parms._response_column+" (model "+k+").");
+
+        if (require_consistent_training_frames) {
+          if (trainingFrameRows < 0) trainingFrameRows = _parms.train().numRows();
+          Frame aTrainingFrame = aModel._parms.train();
+          if (trainingFrameRows != aTrainingFrame.numRows())
+            throw new H2OIllegalArgumentException("Base models are inconsistent: they use different size (number of rows) training frames."
+                    +" Found: "+trainingFrameRows+" (StackedEnsemble) and "+aTrainingFrame.numRows()+" (model "+k+").");
+        }
 
-          // If we don't have a fold_column require nfolds > 1
-          if (aModel._parms._fold_column == null && aModel._parms._nfolds < 2)
-            throw new H2OIllegalArgumentException("Base model does not use cross-validation: " + aModel._parms._nfolds);
+        if (cv_required_on_base_model) {
 
-          // NOTE: we already check that the training_frame checksums are the same, so
-          // we don't need to check the Vec checksums here:
-          if (aModel._parms._fold_column != null &&
-              ! aModel._parms._fold_column.equals(basemodel_fold_column))
-            throw new H2OIllegalArgumentException("Base models are inconsistent: they use different fold_columns.");
+          if (aModel._parms._fold_assignment != basemodel_fold_assignment
```

Comment:
@seb-h2o what if we have following situation:
```
aModel._parms._fold_assignment == Random && basemodel_fold_assignment == AUTO
```
shouldn't we exclude it as well? or it is impossible to get into this state?

Reply:
L376:
` if (basemodel_fold_assignment == AUTO) basemodel_fold_assignment = Random;`

Reply:
@seb-h2o I see now. It was not on the surface let's say :)